code/modules/power/pt_laser.dm

```diff
@@ -76,6 +84,59 @@
 
 	..()
 
+/obj/machinery/power/pt_laser/proc/_enable_input_mechchomp()
+	src.charging = TRUE
+
+/obj/machinery/power/pt_laser/proc/_disable_input_mechchomp()
+	src.charging = FALSE
+
+/obj/machinery/power/pt_laser/proc/_set_input_mechchomp(var/datum/mechanicsMessage/inp)
+	if(!length(inp.signal)) return
+	var/newinput = text2num(inp.signal)
+	if (newinput != src.chargelevel && isnum_safe(newinput) && newinput > 0)
+		src.chargelevel = newinput
+		// Working backwards to update the ui based on the power we've set up.
+		if(chargelevel < 1 KILO WATT)
+			src.input_multi = 1 WATT
+		else if(chargelevel < 1 MEGA WATT)
+			src.input_multi = 1 KILO WATT
+		else if(chargelevel < 1 GIGA WATT)
+			src.input_multi = 1 MEGA WATT
+		else if(chargelevel < 1 TERA WATT)
+			src.input_multi = 1 GIGA WATT
+		else
+			src.input_multi = 1 TERA WATT
+		src.input_number = clamp((src.chargelevel/src.input_multi), 0, src.max_dial_value)
+
+/obj/machinery/power/pt_laser/proc/_enable_output_mechchomp()
+	src.online = TRUE
+	src.process(1)
```

Comment:
This makes it so spamming the signal repeatedly rolls the melt prob much faster than it should, process should only be called by the process controller (usually (kinda))